### PR TITLE
chore: Organize Application-related defaults/constants in the Application API's package.

### PR DESCRIPTION
**Pull request by sbose78**
Hi,

This PR attempts to re-organize code to follow Golang code design recommendations[1]:

* constants/defaults which belong to the Application API needs to live with the API's package.
* Moved the Repository API , AppProject API to its own files for better readability. 
* The `commons` package is becoming an 'assortment of unrelated constants' :( This PR attempts to reduce that bit-by-bit.

https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common

---

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: pkg/apis/application/v1alpha1/repository_types.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
+package v1alpha1
+
+import (
+	"net/url"
+	"regexp"
+
```
Comment (sbose78) [inline pkg/apis/application/v1alpha1/repository_types.go:6]:
```
Check failure on line 6 in pkg/apis/application/v1alpha1/repository_types.go

GitHub Actions
/ Lint Go code
pkg/apis/application/v1alpha1/repository_types.go#L6
File is not `goimports`-ed with -local github.com/argoproj/argo-cd (goimports)
```

No clue what this means :(

Comment (jannfis) [reply pkg/apis/application/v1alpha1/repository_types.go:6]: It means that imports are not ordered as the linter would have expected. The linter assumes the imports come in three separate blocks, each of which is alphabetically sorted:

* Block 1: Golang standard library imports (e.g. `net/http`)
* Block 2: Imports from same module (e.g. `github.com/argoproj/argo-cd/util/cert`)
* Block 3: All other external package imports (e.g. `log "github.com/sirupsen/logrus"`)

All blocks should be separated by one blank line. So, in your case, there's just a blank line missing between the following two lines:

```
        "github.com/argoproj/argo-cd/util/helm"
	log "github.com/sirupsen/logrus"
```

:)

**Comment by jannfis** [comment]
Thank you @sbose78, this is a long overdue refactor imho :)

File: pkg/apis/application/v1alpha1/repository_types.go (hunk 1 above)
Comment (sbose78) [reply pkg/apis/application/v1alpha1/repository_types.go:6]: 😱 Thank you so much!

File: pkg/apis/application/v1alpha1/app_project_types.go [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+
+import (
+	fmt "fmt"
+	"sort"
+	"strconv"
+	strings "strings"
+
+	"github.com/argoproj/argo-cd/util/git"
+	"github.com/argoproj/argo-cd/util/glob"
+	"github.com/google/go-cmp/cmp"
```
Comment (jannfis) [inline pkg/apis/application/v1alpha1/app_project_types.go:11]: Same here, blank line missing :)

**Review by jannfis — APPROVED** [review]
LGTM, I fully approve this refactor and appreciate that you took the time to do it, @sbose78

I will mark this change as breaking, since third-party tools that use Argo CD as Go module might have some work ahead when updating their dependencies.

Requesting review also from @alexmt and @jessesuen

**Comment by sbose78** [comment]
@alexmt Would this be a better time to visit this change now that v2.0.0 is out of the door :)

**Comment by alexmt** [comment]
Thank you @sbose78 . Great change . Agree with @jannfis  that we should've done it a long time ago.

**Review by alexmt — APPROVED** [review]
LGTM

Merging after conflicts are resolved

**Comment by alexmt** [comment]
@sbose78 , don't want to lose this PR. Do you have time to resolve conflicts until the master diverged too much?
Please, let me know if you are busy. I can help to make changes and push to your branch

**Comment by sbose78** [comment]
I can check and re-do if needed :-) Thanks!

File: pkg/apis/application/v1alpha1/repository_types.go [hunk 3]
```diff
@@ -187,12 +190,6 @@ func (repo *Repository) GetHelmCreds() helm.Creds {
 	}
 }
 
-// IsHTTPSURL returns true if supplied URL is HTTPS URL
-func IsHTTPSURL(url string) bool {
```
Comment (sbose78) [inline pkg/apis/application/v1alpha1/repository_types.go]: While this got removed in a different PR, I see another dependency has been introduced 

```
func getCAPath(repoURL string) string {
	if git.IsHTTPSURL(repoURL) {
		if parsedURL, err := url.Parse(repoURL); err == nil {
			if caPath, err := cert.GetCertBundlePathForRepository(parsedURL.Host); err == nil {
				return caPath
			} else {
				log.Warnf("Could not get cert bundle path for host '%s'", parsedURL.Host)
			}
		} else {
			// We don't fail if we cannot parse the URL, but log a warning in that
			// case. And we execute the command in a verbatim way.
			log.Warnf("Could not parse repo URL '%s'", repoURL)
		}
	}
	return ""
}
```
I will take care of it in a different PR.

**Comment by sbose78** [comment]
@alexmt  This is good to go 👍

**Pull request merged**